eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/jpa/JpaEventStorageEngine.java

```diff
@@ -152,6 +152,35 @@ protected static <T> DomainEventMessage<T> asDomainEventMessage(EventMessage<T>
                : new GenericDomainEventMessage<>(null, event.getIdentifier(), 0L, event, event::getTimestamp);
     }
 
+    /**
+     * Returns a batch of event data as object entries in the event storage with a
+     * greater than the given {@code token}.Size of event is decided by batchSize()
+     *
+     * @param token     Object describing the global index of the last processed event.
+     * @return A batch of event messages as object stored since the given tracking token.It
+     */
+    protected List<Object[]> fetchEvents(GapAwareTrackingToken token) {
+        TypedQuery<Object[]> query;
+        if (token == null || token.getGaps().isEmpty()) {
+            query = entityManager().createQuery(
+                    "SELECT e.globalIndex, e.type, e.aggregateIdentifier, e.sequenceNumber, e.eventIdentifier, "
+                            + "e.timeStamp, e.payloadType, e.payloadRevision, e.payload, e.metaData " +
+                            "FROM " + domainEventEntryEntityName() + " e " +
+                            "WHERE e.globalIndex > :token ORDER BY e.globalIndex ASC", Object[].class);
+        } else {
+            query = entityManager().createQuery(
+                    "SELECT e.globalIndex, e.type, e.aggregateIdentifier, e.sequenceNumber, e.eventIdentifier, "
+                            + "e.timeStamp, e.payloadType, e.payloadRevision, e.payload, e.metaData " +
+                            "FROM " + domainEventEntryEntityName() + " e " +
+                            "WHERE e.globalIndex > :token OR e.globalIndex IN :gaps ORDER BY e.globalIndex ASC",
+                    Object[].class
+            ).setParameter("gaps", token.getGaps());
+        }
+        return query.setParameter("token", token == null ? -1L : token.getIndex())
```

Comment:
Nit: The indentation is slightly off here. I would expect `.setMaxResults(batchSize())` to start at the same point as `.setParamater(...)`.

Reply:
I have amended them.